### ISSUE TITLE
[bitnami/nginx-ingress-controller] 9.3.27+ duplicates location / block

### ISSUE THREAD

**Issue by steache** [bitnami/containers#23525]
### Name and Version

nginx-ingress-controller:9.3.27

### What steps will reproduce the bug?

1. install nginx-ingress-controller:9.3.27
2. create ingress resource
3. nginx will be unable to reload, because nginx.conf is broken

for each
`server {` there will be multiple
`location / {`

### What is the expected behavior?

only one location should be created

### What do you see instead?

multiple location / notations

### Additional information

9.3.28 is also affected, last stable was 9.3.26

thank you  for any help :) I will try to look for any changes in:

- [[bitnami/nginx-ingress-controller] Release 1.6.1-debian-11-r0](https://github.com/bitnami/containers/commit/6daaa2d498a8958eaa2ce5816ed7eda08f455b78)
- [[bitnami/nginx-ingress-controller] Release 1.6.1-debian-11-r1](https://github.com/bitnami/containers/commit/a3e40410429479808f8456d79001d10ba965225c)
- [[bitnami/nginx-ingress-controller] Release 1.6.1-debian-11-r2](https://github.com/bitnami/containers/commit/e0fd2358be3764a91b95b8bcb944a17bf63cc8fb)
- [[bitnami/nginx-ingress-controller] Release 1.6.2-debian-11-r0](https://github.com/bitnami/containers/commit/b0880937d5e05e99b0f92b0d2d4121802aa8a478)

**Comment by dgomezleon** [comment]
Hello @steache,

This version was released with this [PR](https://github.com/bitnami/charts/pull/14712/files). 

Could you confirm if the issue you are facing is related to nginx-ingress-controller `1.6.1-debian-11-r0` or nginx `1.22.1-debian-11-r35`? Could you also share the ingress resource and the resulting `nginx.conf` file?

**Comment by kallaics** [comment]
Hi @dgomezleon ,
We are facing with same issue and I hope our information are also will be helpful.

Latest version without issues: 9.3.26
Same issue with releases: 9.3.27 and 9.3.28 as well.

Reply for your question:
Deployment version: 9.3.28
Image version: docker.io/bitnami/nginx-ingress-controller:1.6.2-debian-11-r0

Our root cause: We found for one ingress definition with 2 different configuration in same file. The first one was empty and the second contains the real definition. So it looks like the templating are generates one configuration set with "empty" values and the second templating will be the real configuration inside same file. That is why I assume it can be a templating issue.

I hope it will be helpful.

**Comment by dgomezleon** [comment]
Hi @kallaics,

Thanks for the feedback.

I was not able to reproduce the issue. Could you please share with us the ingress resource you created and the difference and the generated `nginx.conf` file?

**Comment by kallaics** [comment]
Hi @dgomezleon ,
Sorry about a late reply. Yesterday was a typical Monday :D
I removed some sensitive data.

My ingress configuration:

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    meta.helm.sh/release-name: prometheus-stack
    meta.helm.sh/release-namespace: monitoring
    nginx.ingress.kubernetes.io/auth-realm: Authentication Required
    nginx.ingress.kubernetes.io/auth-secret: prometheus-federation-auth
    nginx.ingress.kubernetes.io/auth-type: basic
    nginx.ingress.kubernetes.io/whitelist-source-range: 10.20.30.0/16
  labels:
    app: kube-prometheus-stack-prometheus
    app.kubernetes.io/instance: prometheus-stack
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: kube-prometheus-stack
    app.kubernetes.io/version: 45.0.0
    chart: kube-prometheus-stack-45.0.0
    helm.toolkit.fluxcd.io/name: prometheus-stack
    helm.toolkit.fluxcd.io/namespace: monitoring
    heritage: Helm
    release: prometheus-stack
  name: prometheus-stack-kube-prom-prometheus
  namespace: monitoring
spec:
  ingressClassName: nginx-ic
  rules:
  - host: prometheus.domain.tld
    http:
      paths:
      - backend:
          service:
            name: prometheus-stack-kube-prom-prometheus
            port:
              number: 9090
        path: /
        pathType: ImplementationSpecific
```

The generated config file 
The issue is probably with the empty definition between lines 239 and 480.

[nginx-cfg.txt](https://github.com/bitnami/containers/files/10730338/nginx-cfg.txt)

Please let me know, if you need more details and thanks for your support.

**Comment by dgomezleon** [comment]
Thanks for the feedback again.

It is clearly something introduced in nginx-ingress-controller 1.6.1. Yo can fin the issue [here](https://github.com/kubernetes/ingress-nginx/issues/9569)

I tested it with ` tag: 1.6.0-debian-11-r10` without any issues. You could apply this as workaround.

We will work on releasing version 1.6.3 ASAP (we need to create an upstream issue since it is not tagged).

**Comment by kallaics** [comment]
You are welcome. 

The mentioned issue is exactly the same, that are happen on our ingress controllers as well.

Yes, I can confirm since release 1.6.1 we had issues with it, but I didn't found the reported issue, that you mentioned.

Thanks for all of your efforts to the new release. 👍

**Comment by dgomezleon** [comment]
Hello,

A new version of the nginx-ingress-cotroller container has been released (we are still working on the charts).  Could you try using this new image `1.6.4-debian-11-r0`?

**Comment by kallaics** [comment]
Hi @dgomezleon ,

I tested with the new image and I can confirm it works well now. 
Thanks for your efforts and the quick fix.

**Comment by steache** [comment]
fixed in nginx-ingress-controller@9.3.29, thank you so much! :)
